config/configstore/bootstrapper.go

```diff
@@ -25,15 +27,44 @@ func (*Bootstrapper) Bootstrap(context map[string]interface{}) error {
 	if !ok {
 		return errors.New("identity factory service not initialised")
 	}
+
+	idFactoryV2, ok := context[identity.BootstrappedDIDFactoryV2].(identity.FactoryInterface)
+	if !ok {
+		return errors.New("configstore: identity factory not initialised")
+	}
+
 	idService, ok := context[identity.BootstrappedDIDService].(identity.Service)
 	if !ok {
 		return errors.New("identity service not initialised")
 	}
 
+	dispatcher, ok := context[jobsv2.BootstrappedDispatcher].(jobsv2.Dispatcher)
+	if !ok {
+		return errors.New("dispatcher not initialised")
+	}
+
+	ethClient, ok := context[ethereum.BootstrappedEthereumClient].(ethereum.Client)
+	if !ok {
+		return errors.New("ethereum client not initialised")
+	}
+
 	repo := &repo{configdb}
-	service := &service{repo, idFactory, idService, func() ProtocolSetter {
-		return context[bootstrap.BootstrappedPeer].(ProtocolSetter)
-	}}
+	service := &service{
+		repo:      repo,
+		idFactory: idFactory,
+		idService: idService,
+		protocolSetterFinder: func() ProtocolSetter {
+			return context[bootstrap.BootstrappedPeer].(ProtocolSetter)
+		},
+		dispatcher:  dispatcher,
+		idFactoryV2: idFactoryV2,
+	}
+
+	go dispatcher.RegisterRunner(generateIdentityRunnerName, generateIdentityRunner{
```

Comment:
Is `RegisterRunner` a fast-executing function? should this be non-blocking? Since our health-check at startup will not know about when is this step done, might confuse the readiness.